js/gateio.js

```diff
@@ -622,7 +622,14 @@ module.exports = class gateio extends Exchange {
                 for (let i = 0; i < response.length; i++) {
                     const market = response[i];
                     const id = this.safeString (market, 'name');
-                    const [ baseId, quoteId, date ] = id.split ('_');
```

Comment:
PHP gets an error when there's no date, because these too many values to unpack

Reply:
In such cases where one part may be missing, we should replace the value-unpacking line and split it into three lines:

```JavaScript
const parts = id.split ('_');
const baseId = this.safeString (parts, 0);
const quoteId = this.safeString (parts, 1);
const date = this.safeString (parts, 2);
```